CONTRIBUTING.md

```diff
@@ -23,3 +23,59 @@ use GitHub pull requests for this purpose. Consult [GitHub Help] for more
 information on using pull requests.
 
 [GitHub Help]: https://help.github.com/articles/about-pull-requests/
+
+## Building Dagger
+
+Dagger is built with [`bazel`](https://bazel.build).
+
+### Setting up Bazel
+
+* Go to [http://bazel.build](http://bazel.build), click on `GET BAZEL`, and install Bazel.
+* In *Android Studio*, go to `Preferences > Plugins`.
+  * search for `bazel` and install the plugin.
+  * If no result shows up, click on `Search in repositories`, search for `bazel` and install the plugin.
+  * If you still get no result, then the bazel plugin isn't available for your version of Android Studio.
+    * Go to [Integrating Bazel with IDEs](https://docs.bazel.build/versions/master/ide.html), click on `Android Studio plugin`.
+	* The `COMPATIBILITY` column indicates which versions of Android Studio have an available Bazel plugin.
+	* You can find the corresponding numbering scheme in the file names of [Android Studio downloads](https://developer.android.com/studio/#downloads). You may need to [download a Preview](https://developer.android.com/studio/preview/).
+
+### Building Dagger from the command line
+
+* Clone the Dagger repository:
+
+```
+git clone git@github.com:google/dagger.git dagger
+```
+
+*  Build the Dagger project:
+
+```
+bazel build
+```
+
+* You can install the Dagger libraries in your local maven repository with the `install-local-snapshot` script.
+* It will build the libraries and install them with a `LOCAL-SNAPSHOT` version.
+* If you get the following error: `ERROR: missing input file '@androidsdk//:build-tools/26.0.2/aapt'`.
+  * Install the missing build tools version: `android --use-sdk-wrapper update sdk -u --filter build-tools-26.0.2`
+
+```
+./util/install-local-snapshot.sh
+```
+
+* To run tests:
+
+```
+bazel test //...
```

Comment:
I'd like to beef up this section, I actually have not figured out how to run any test at all. How does one run all tests / run a subset of tests? Bazel docs are full of many words which I don't care about :) .

Reply:
Running `bazel test //...` shows the names of the tests passing, so that helped figure out that I could do something like `bazel test //javatests/dagger/internal/codegen:KeyFactoryTest`

Still, ideally, would be nice to link to a doc that explains those naming rules.

Reply:
I'm not sure you need to teach Bazel here.

The naming is pretty straightforward it's just path and class name. General targets are https://docs.bazel.build/versions/master/build-ref.html

Reply:
Hopefully I'm not teaching Bazel with just two command lines examples. Contributors to Dagger shouldn't have to start by learning how bazel works, so giving them the things they need the most and the link to that guide seems important.

This whole section might seem painfully obvious, but when you've never used Bazel (as most non googlers) it's a barrier to contributing. Let's lower it.

Reply:
How to run individual tests is too low